cirq/work/observable_measurement_data.py

```diff
@@ -0,0 +1,478 @@
+
+    @property
+    def n_repetitions(self):
+        return len(self.bitstrings)
+
+    @property
+    def results(self):
+        """Yield individual setting results as `ObservableMeasuredResult`
+        objects."""
+        for setting in self._simul_settings:
+            yield ObservableMeasuredResult(
+                setting=setting,
+                mean=self.mean(setting),
+                variance=self.variance(setting),
+                repetitions=len(self.bitstrings),
+                circuit_params=self._meas_spec.circuit_params,
+            )
+
+    @property
+    def records(self):
+        """Yield individual setting results as dictionary records.
+
+        This is suitable for passing to pd.DataFrame constructor, perhaps
+        after chaining these results with those from other BitstringAccumulators.
+        """
+        for result in self.results:
+            record = dataclasses.asdict(result)
+            del record['circuit_params']
+            record.update(**self._meas_spec.circuit_params)
+            yield record
+
+    def _json_dict_(self):
+        from cirq.study.result import _pack_digits
+
+        def ndarray_to_hex_str(a):
+            return _pack_digits(a, pack_bits='never')[0]
+
+        return {
+            'cirq_type': self.__class__.__name__,
+            'meas_spec': self.meas_spec,
+            'simul_settings': self.simul_settings,
+            'qubit_to_index': list(self.qubit_to_index.items()),
+            'bitstrings': ndarray_to_hex_str(self.bitstrings),
+            'chunksizes': ndarray_to_hex_str(self.chunksizes),
+            'timestamps': ndarray_to_hex_str(self.timestamps),
+        }
+
+    @classmethod
+    def _from_json_dict_(
+        cls,
+        *,
+        meas_spec,
+        simul_settings,
+        qubit_to_index,
+        bitstrings,
+        chunksizes,
+        timestamps,
+        **kwargs,
+    ):
+        from cirq.study.result import _unpack_digits
+
+        def hex_str_to_ndarray(hexstr):
+            # When binary=False, the other arguments are not needed.
+            return _unpack_digits(hexstr, binary=False, dtype=None, shape=None)
+
+        return cls(
+            meas_spec=meas_spec,
+            simul_settings=simul_settings,
+            qubit_to_index=dict(qubit_to_index),
+            bitstrings=hex_str_to_ndarray(bitstrings),
+            chunksizes=hex_str_to_ndarray(chunksizes),
+            timestamps=hex_str_to_ndarray(timestamps),
+        )
+
+    def __eq__(self, other):
+        if not isinstance(other, BitstringAccumulator):
+            return False  # coverage: ignore
+
+        if (
+            self.max_setting != other.max_setting
+            or self.simul_settings != other.simul_settings
+            or self.circuit_params != other.circuit_params
+            or self.qubit_to_index != other.qubit_to_index
+        ):
+            return False  # coverage: ignore
```

Comment:
with `cirq.testing.EqualsTester` it should be easy to cover these cases instead of coverage:ignore-ing them.

Reply:
Looking into this

Reply:
Sounds good - these are the last ones, right?